internal/pkg/remote/remote.go

```diff
@@ -266,13 +278,16 @@ func (c *Config) Logout(uri string) error {
 	if err := credential.Manager.Logout(uri); err != nil {
 		return err
 	}
-	for i, cred := range c.Credentials {
+	// Older versions of Singularity can create duplicate entries with same URI,
+	// so loop must handle removing multiple matches (#214).
+	for i := 0; i < len(c.Credentials); i++ {
+		cred := c.Credentials[i]
 		if remoteutil.SameURI(cred.URI, uri) {
 			c.Credentials = append(c.Credentials[:i], c.Credentials[i+1:]...)
-			return nil
+			i = -1
 		}
 	}
-	return fmt.Errorf("%s is not configured", uri)
+	return nil
```

Comment:
Note - the `%s not configured` error was redundant here. If `credential.Manager.Logout` succeeds in logging out of something then it's not an error to not remove an entry from `remote.yaml`... We may have successfully cleaned up a mismatch from `remote.yaml` -> `docker-config.json`